esp/smc/SMCLib/TpWrapper.cpp

```diff
@@ -2077,21 +2077,30 @@ class CContainerWUClusterInfo : public CSimpleInterfaceOf<IConstWUClusterInfo>
     StringAttr thorQueue;
     ClusterType platform;
     unsigned clusterWidth;
+    StringArray thorProcesses;
+    unsigned roxieRedundancy = 2;
+    unsigned channelsPerNode = 1;
 
 public:
-    CContainerWUClusterInfo(const char* _name, const char* type, unsigned _clusterWidth)
+    CContainerWUClusterInfo(const char* _name, const char* type, unsigned _clusterWidth, IPropertyTree *cfg)
         : name(_name), clusterWidth(_clusterWidth)
     {
         StringBuffer queue;
         if (strieq(type, "thor"))
         {
             thorQueue.set(getClusterThorQueueName(queue.clear(), name));
             platform = ThorLCRCluster;
+            thorProcesses.append(name);
         }
         else if (strieq(type, "roxie"))
         {
             agentQueue.set(getClusterEclAgentQueueName(queue.clear(), name));
             platform = RoxieCluster;
+            if (cfg)
+            {
+                roxieRedundancy = (unsigned) cfg->getPropInt("@replicas", roxieRedundancy);
+                channelsPerNode = (unsigned) cfg->getPropInt("@numChannels", channelsPerNode);
```

Comment:
numChannels is total number of channels in the cluster, not per node

channels per node is always 1 in containerized system. frequently 2 but could be more in bare metal